aerogearcatalog/unifiedpush-apb
# Update binding error handling

JIRA: https://issues.jboss.org/browse/AEROGEAR-2628

Changes:

1. Blocked the user to create a second variant - this used to override the previous one and we do not want this behavior. The binding in Openshift will fail with an error from the broker, the user can then delete the binding secret only and not view it. @pb82  This can be removed, but it shows the user that an issue has occurred in the binding/variant creation.
This is an example of the UI after one successful and one failed binding 
![image](https://user-images.githubusercontent.com/7661017/39479142-3e0a470a-4d5c-11e8-82e2-e0e9fb7ed5a1.png)

2. The binding/variant creation will fail for iOS when some bad info is sent (invalid cert or wrong passphrase) - this will also return an error from the broker 

Testing:

To test (use cordova )

- Create an android binding it should be successful - use mobile cli to note the variant info
- Try to create a second android - it should fail and allow the user to delete the secret in UI - use the mobile cli to verify the variant did not change/update

- Create an iOS binding with bad cert and the wrong passphrase - it should fail and UI should display error and option to delete secret only

- Create a valid iOS binding and it should succeed - use cli to verify this
- Try to create a second iOS binding and it should fail as per the android test
